data_model/src/transaction.rs

```diff
@@ -320,6 +320,29 @@ impl SignedTransaction {
     }
 }
 
+#[cfg(feature = "transparent_api")]
+impl TryFrom<(SignaturesOf<TransactionPayload>, TransactionPayload)> for SignedTransaction {
+    type Error = &'static str;
```

Comment:
`iroha_data_model` is a highly-domain-specific module, thus it's better to use stricter, domain-focused errors.

In other words, please define this error as an `enum` deriving `thiserror::Error`.